### PR TITLE
Update liquidsoap and dependencies to 2.1.2

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Description of changes

https://github.com/savonet/liquidsoap/releases/tag/v2.1.0
https://github.com/savonet/liquidsoap/releases/tag/v2.1.1
https://github.com/savonet/liquidsoap/releases/tag/v2.1.2

https://github.com/mmottl/pcre-ocaml/releases/tag/7.5.0

https://github.com/savonet/ocaml-taglib/releases/tag/v0.3.10

https://github.com/savonet/ocaml-ffmpeg/releases/tag/v1.1.5
https://github.com/savonet/ocaml-ffmpeg/releases/tag/v1.1.6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: pkgs/development/ocaml-modules/pcre/default.nix [hunk 1]
```diff
@@ -2,15 +2,15 @@
 
 buildDunePackage rec {
   pname = "pcre";
-  version = "7.4.6";
+  version = "7.5.0";
 
   useDune2 = true;
 
-  minimalOCamlVersion = "4.08";
+  minimalOCamlVersion = "4.12";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/pcre/default.nix:9]: This change removes quite a few packages (for OCaml < 4.12). How bad is this?

Comment (dali99) [reply pkgs/development/ocaml-modules/pcre/default.nix:9]: in nixpkgs it seems fine, for external consumers that might be different (though that's always different).

The only alternative is packaging the old one as ocaml_pcre_746 or something. But that will still break.

I'm not sure why the ocaml subsystem in nixpkgs "supports" such a wide version range anyways though. It's quite different from how other languages in nixpkgs do things.

I might be missing some details but having a policy for generally only supporting the last two minor versions similar to python seems reasonable to me.

Anyways, should I be bumping the version constraint for the library consumers as well in this case?

Comment (vbgl) [reply pkgs/development/ocaml-modules/pcre/default.nix:9]: Thanks for your reply.

> Anyways, should I be bumping the version constraint for the library consumers as well in this case?

I think it’s fine.

**Review by nagisa — APPROVED** [review]
I’ve tested this extensively and it works well.

**Pull request merged**